draft-decraeneginsberg-lsr-isis-fast-flooding.xml

```diff
@@ -276,55 +274,57 @@
+      <t>However, the flooding rate has not been fundamentally altered.</t>
     </section>
 
 	
 
 
 		<section anchor="FloodingTLV" title="Flooding Parameters TLV">
-			<t>This document defines a new TLV called "Flooding Parameters TLV" that may be included in IIH or PSNP PDUs. It allows the LSP receiver to advertise flooding related parameters and capabilities which allows the LSP sender to better adapt to the receiver.</t>
-			<t>Type: TBD1.</t>
-			<t>Length: variable, the size in octet of the Value field.</t>
-			<t>Value: a list of sub-TLVs.</t>
+			<t>This document defines a new Type-Length-Value tuple (TLV) called the "Flooding Parameters TLV" that may be included in IS to IS Hello (IIH) or Partial Sequence Number PDUs (PSNPs). It allows the LSP receiver to advertise flooding related parameters and capabilities and allows the LSP sender to better adapt to the receiver.</t>
+			<t>Type: TBD1</t>
+			<t>Length: variable, the size in octets of the Value field</t>
+			<t>Value: a list of sub-TLVs</t>
 			<t>Two sub-TLVs are defined in this document. The support of any sub-TLV is OPTIONAL.</t>
-			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The IS transmitter uses the latest received value for each parameter (sub-TLV) until a new value is advertised by the IS receiver. Note however that IIH and PSNP are not reliably exchanged, hence may never be received. For a parameter which has never been advertised, the IS transmitter use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
-			<section anchor="LSPReceptionWindow" title="InterfaceLSPReceiveWindow sub-TLV">
-				<t>The sub-TLV InterfaceLSPReceiveWindow advertises the maximum number of un-acknowledged LSPs that the node can receive/process with no separation interval between LSPs. This is meant to be the size of the LSP input queue of the IS-IS process in the control plane.</t>
-				<t>Type: 1.</t>
-				<t>Length: 4 octets.</t>
-				<t>Value: number of un-acknowledged LSPs which can be sent back to back.</t>
-				<t>Note that if an LSP has not been acknowledged and is sent again, it does not count twice. The reason is that this LSP is assumed to be lost and hence not in a buffer at the receiver.</t>
+			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The LSP transmitter uses the latest received value for each parameter until a new value is advertised by the LSP receiver. However, IIHs and PSNPs are not reliably exchanged, and may never be received, so parameters MAY be sent even if there is no change in value since the last transmission. For a parameter which has never been advertised, the LSP transmitter SHOULD use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
```

Comment:
I think the parameters will need to be sent in every hello since the
sender never knows when/if the receiver has received them.
It could possibly be sent only on change in a PSNP as a means of optimizing
the signaling (i.e., no immediate Hello need to be sent).
If we agree, we need to reword.

Also remove the word "unacknowledged".

Reply:
Regarding hellos, I don't think that we need to force the advertiser to include the TLV in all hellos. It could be left to its choice. (and it can assume that at some points IIH are receiver otherwise the adjacency would go down. Also if the parameters are static, advertising them during the 3-way handshake is probably enough).
However, if you want to mandate the sending in all IIH, it also works for me.
The main issue if that if you want to use the TLV in both IIH and PSNP, you would need to change the TLV in both. This may be less convenient for implementations sending IIH in a distributed way (on the line card)

Reply:
We don't need to force TLV advertisements. We've already made the case for why it should be done with some degree of redundancy. I propose the compromise of upgrading from 'MAY' to 'SHOULD'.

Reply:
Removed two instances of 'un-acknowledged'.